### ISSUE TITLE
ulimit settings required for ldas_setup on Discover

### ISSUE THREAD

**Issue by amfox37** [GEOS-ESM/GEOSldas#607]
As a new user on Discover I was having a somewhat cryptic segfault when running ldas_setup as this is done on a login node and I didn't have:
```
ulimit -s unlimited
ulimit -v unlimited
```
in my .bashrc
Was wondering if we should somehow incorporate these settings into ldas_setup, or at least add a warning about this to the [readme](https://github.com/GEOS-ESM/GEOSldas#readme)? 
Or do we just expect folks to already have this in their .bashrc? 
What do you think @weiyuan-jiang ?

**Comment by weiyuan-jiang** [comment]
@mathomp4  Where do you think it is a better place for this?  g5_modules or rc setting files?

**Comment by mathomp4** [comment]
@weiyuan-jiang We usually put this in the script being run itself. For GEOSgcm, `gcm_setup` doesn't really do anything (pure csh), but in `gcm_run.j` we have `limit stacksize unlimited` (as it's a csh script as well).

The fun is that `ldas_setup` is python now, right? But it is calling commands with subprocess. If what it is calling is itself a shell script, then you'd want to put the right `limit` or `ulimit` calls in those scripts.

If you are calling an executable directly, you might have to do something like:

```python
import resource
resource.setrlimit(resource.RLIMIT_STACK, (resource.RLIM_INFINITY, resource.RLIM_INFINITY))
```

I think this is the [Pythonic equivalent](https://docs.python.org/3.9/library/resource.html) if `ulimit -s unlimited`. I'm not sure what the virtual memory one is. Maybe `resource.RLIMIT_VMEM` or `resource.RLIMIT_AS`?

@amfox37 Do you know what command `ldas_setup` was trying to run when you got the error?

**Comment by amfox37** [comment]
My understanding is I was getting a seg fault when preprocess_ldas.x  called with c_f2g on line 726 enters read_cat_param

**Comment by mathomp4** [comment]
@amfox37 Okay. Makes sense. Perhaps if the `setrlimit` command was done at the top of `ldas_setup` that might be enough? I don't think it needs to be done before every call.

**Comment by gmao-rreichle** [comment]
Addressed with #620
